platform/commonUI/edit/src/actions/RemoveAction.js

```diff
@@ -80,21 +82,44 @@ define(
                 var persistence = domainObject.getCapability('persistence');
                 return persistence && persistence.persist();
             }
+            
+            // Checks current object and ascendants of current
+            // object with object being removed, if the current
+            // object or any in the current object's path is being removed,
+            // navigate back to parent of removed object. 
+            function checkObjectNavigation(object, parentObject) {
+                // Traverse object starts at current location
+                var traverseObject = (navigationService).getNavigation();
+                
+                // Stop at ROOT of folder path
+                while (traverseObject.getId() !== ROOT_ID) {
```

Comment:
We could remove the need to explicitly look for the root ID by just checking if traverseObject is defined (or even just truthy), e.g. `while (traveseObject) {`